tests/fixtures/endpoints_handler.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A class to register mock endpoint functions in a similar fashion to FastAPI with
+class methods that can process an incoming request with a dynamic url pattern"""
+
+import inspect
+import logging
+import re
+from functools import partial
+from typing import Any, Callable
+
+import httpx
+from pydantic import BaseModel
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class MatchableEndpoint(BaseModel):
+    """Endpoint data with the url turned into regex string to get parameters in path"""
+
+    url_pattern: str
+    endpoint_function: Callable
+
+
+class EndpointsHandler:
+    """
+    A class used to register mock endpoints with decorators similar to FastAPI.
+    Tag endpoint functions with EndpointHandler.get("/some/url-with/{variables}").
+    .post and .patch are also implemented at the moment.
+    The regex compiler function will turn the url specified in the decorator function
+    into a regex string capable of capturing the variables in the url (curly brackets)
+    with named groups. That in turn enables linking the named path variables to the
+    variables in the endpoint function itself.
+    """
+
+    methods: dict[str, list[MatchableEndpoint]] = {
+        "GET": [],
+        "POST": [],
+        "PATCH": [],
+    }
+
+    @staticmethod
+    def compile_regex_url(url_pattern: str):
```

Comment:
```suggestion
    def _compile_regex_url(url_pattern: str):
```

Reply:
Changed in [f9a5cf1](https://github.com/ghga-de/ghga-connector/pull/68/commits/f9a5cf199803a93f4f8f26826410e0c883a01d7b)